frame/curve-amm/src/lib.rs

```diff
@@ -484,6 +504,105 @@ pub mod pallet {
 
 			Ok(())
 		}
+		fn exchange(
+			who: &Self::AccountId,
+			pool_id: PoolId,
+			i: PoolTokenIndex,
+			j: PoolTokenIndex,
+			dx: Self::Balance,
+			min_dy: Self::Balance,
+		) -> Result<(), DispatchError> {
+			let prec = T::Precision::get();
+			let zero_b = Self::Balance::zero();
+			ensure!(dx >= zero_b, Error::<T>::WrongAssetAmount);
+
+			let (provider, pool_id, dy) =
+				Pools::<T>::try_mutate(pool_id, |pool| -> Result<_, DispatchError> {
+					let pool = pool.as_mut().ok_or(Error::<T>::PoolNotFound)?;
```

Comment:
Why do we have a `as_mut` here? The `pool` should be a `&mut ...` considering we are using `try_mutate`, unless I missed something?

Reply:
try_mutate() returns &mut Option<T> so then with as_mut it is converted to Option<&mut T>